cmd/wallet-web/src/pages/Dashboard.vue

```diff
@@ -129,52 +150,54 @@ export default {
     ...mapGetters('agent', { getAgentInstance: 'getInstance' }),
     ...mapGetters(['getCurrentUser', 'getAgentOpts']),
     fetchAllCredentials: async function (getCredential) {
-      let { contents } = await getCredential;
+      const { contents } = await getCredential;
       console.log(`found ${Object.keys(contents).length} credentials`);
       const _filter = (id) => {
         return !contents[id].type.some((t) => filterBy.includes(t));
       };
 
-      const _createCard = (id) => {
-        return { content: contents[id], flipped: false };
-      };
+      this.credentials = Object.keys(contents)
+        .filter(_filter)
+        .map((id) => contents[id]);
 
-      this.cards = Object.keys(contents).filter(_filter).map(_createCard);
+      console.debug(`showing ${this.credentials.length} credentials`);
 
-      console.log(`showing ${this.cards.length} credentials`);
-
-      // Reading the values of the credentials and mapping it to the credential display data schemas
-      this.cjson.forEach((obj) => {
-        let flattened = {};
-        for (let credentialKey in this.cards) {
-          this.flatten(this.cards[credentialKey].content, flattened);
-          for (let credentialContent in flattened) {
-            if (
-              obj.credentialSubject.hasOwnProperty(credentialContent) &&
-              obj.schema === this.credDisplayName(this.cards[credentialKey].content)
-            ) {
-              obj.credentialSubject[credentialContent] = flattened[credentialContent];
+      Object.values(this.credentials).map((credential) => {
+        const currentCredentialType = this.getCredentialType(credential);
+        Object.entries(this.supportedCredentials).map(
+          ([supportedCredentialType, supportedCredential]) => {
+            if (supportedCredentialType === currentCredentialType) {
+              const processedCredential = this.getCredentialDisplayData(
+                credential,
+                supportedCredential
+              );
+              this.processedCredentials.push({ ...processedCredential, category: 'manifested' });
```

Comment:
why you need category here?

Reply:
you separate the credential we support and cred we dont , handle generic credential scenarios